### PR TITLE
Replace 'dict_label' argument by 'path'.

### LINKED ISSUE #60: Specification of path within hdf5 file inconsistent between load and save function
Both while loading from and saving to a file, the user can specify paths within the hdf5 file. They are specified with keyword arguments that are called differently in the `load` and `save` function:

`save(filename, d, write_mode='a', overwrite_dataset=False, resize=False, dict_label='', compression=None)`

`load(filename, path='', lazy=False)`

So, we have `dict_path` in the `save` function and `path` in the load function. In my opinion, we should name the arguments consistently across the two function and opt for calling it `path`.

This will of course break user code so we should only include this in the next major release.

### PR DESCRIPTION
**Pull request by mschmidt87**
- Introduce new keyword argument `path` in save function
- I set the default value for `path` to `None` instead of `''` because I find this clearer.
- Add DeprecationWarning if `dict_label` is used.
- Add Exception if `dict_label` and `path` are simultaneously used
- Adapt tests for new argument name and add test for exception

Fixes #60 .

### CODE REVIEW

File: h5py_wrapper/wrapper.py [hunk 1]
```diff
@@ -83,6 +83,15 @@ def save(filename, d, write_mode='a', overwrite_dataset=False,
...
+                warnings.warn("Deprecated argument dict_label provided. "
+                              "`dict_label` will be removed in the next release. "
+                              "Please use `path` instead.",
+                              DeprecationWarning)
+                if path is not None:
+                    raise ValueError("dict_label and path must not "
+                                     "be defined simultaneously.")
+                path = dict_label                
+            if path:
                 base = f.require_group(dict_label)
```
Comment (jakobj) [inline h5py_wrapper/wrapper.py]: shouldn't you use `path` here instead of `dict_label`?

File: h5py_wrapper/wrapper.py [hunk 2]
```diff
@@ -83,6 +83,15 @@ def save(filename, d, write_mode='a', overwrite_dataset=False,
     else:
         try:
             if dict_label:
+                warnings.warn("Deprecated argument dict_label provided. "
+                              "`dict_label` will be removed in the next release. "
```
Comment (jakobj) [inline h5py_wrapper/wrapper.py]: we should be consistent with use of the backtick \`, either we use it around every variable name or around none.

**Comment by mschmidt87** [comment]
Thanks for raising the two points. I fixed the bug in `require_path` and removed the backticks since we haven't used them in any other message. Though I would actually prefer using them for argument and function names. But I don't know the general convention.

**Comment by mschmidt87** [comment]
P.S.: No idea about travis. See #62 .

**Pull request merged**